apps/mobile/src/features/account-selector-sheet.tsx

```diff
@@ -0,0 +1,88 @@
+import { ScrollView } from 'react-native';
+import { useSharedValue } from 'react-native-reanimated';
+
+import { getAvatarIcon } from '@/components/avatar-icon';
+import { Modal } from '@/components/bottom-sheet-modal';
+import { Draggable } from '@/components/draggable';
+import { AccountCard } from '@/components/wallet-settings/account-card';
+import { APP_ROUTES } from '@/constants';
+import { useAccounts, userUpdatesAccountOrder } from '@/state/accounts/accounts.slice';
+import { destructAccountIdentifer, useAppDispatch } from '@/state/utils';
+import { BottomSheetModal, BottomSheetScrollView } from '@gorhom/bottom-sheet';
+import { useRouter } from 'expo-router';
+
+import { Box } from '@leather.io/ui/native';
+
+export function AccountSelectorSheet({ modalRef }: { modalRef: RefObject<BottomSheetModal> }) {
+  const scrollViewRef = useRef<ScrollView>(null);
+  const placeholderIdx = useSharedValue<null | number>(null);
+  const accounts = useAccounts().list;
+  const direction = useSharedValue<'down' | 'up'>('down');
+  const router = useRouter();
+  const dispatch = useAppDispatch();
+  const checkIdxWithinBounds = useCallback(
+    (id: number) => {
+      return id >= 0 && id < accounts.length;
+    },
+    [accounts.length]
+  );
+
+  const accountIds = useMemo(() => accounts.map(acc => acc.id), [accounts]);
+  const swapAccountIndexes = useCallback(
+    (idx1: number | null, idx2: number | null) => {
+      if (
+        idx1 === null ||
+        idx2 === null ||
+        !checkIdxWithinBounds(idx1) ||
+        !checkIdxWithinBounds(idx2)
+      )
+        return;
+      const temp = accountIds[idx1];
+      /** @ts-expect-error we know that both indexes have both value in accounts array */
+      accountIds[idx1] = accountIds[idx2];
+      /** @ts-expect-error we know that both indexes have both value in accounts array */
+      accountIds[idx2] = temp;
+      dispatch(userUpdatesAccountOrder({ accountIds }));
+    },
+    [accountIds, checkIdxWithinBounds, dispatch]
+  );
+  const onAccountPress = useCallback(
+    (accountId: string) => {
+      const { fingerprint, accountIndex } = destructAccountIdentifer(accountId);
+      modalRef.current?.close();
+      router.navigate({
+        pathname: APP_ROUTES.WalletWalletsSettingsConfigureAccount,
+        params: { wallet: fingerprint, account: accountIndex },
+      });
+    },
+    [router, modalRef]
+  );
+
+  return (
+    <Modal shouldHaveContainer={false} ref={modalRef}>
+      <BottomSheetScrollView ref={scrollViewRef}>
+        <Box p="5" gap="5">
+          {accounts.map((account, idx) => {
+            return (
+              <Draggable
+                idx={idx}
+                direction={direction}
+                scrollViewRef={scrollViewRef}
+                placeholderIdx={placeholderIdx}
+                accountsLength={accounts.length}
+                key={account.id}
+                accountId={account.id}
+                Component={() => (
```

Comment:
Curious on this bc I keep seeing it, we never have captialized props in the extension (even if we are passing ReactNodes), is that typ in RN code? Not opposed to it, but I think in the icon work I refactored some to lowercase `icon` out of habit.

Reply:
This looks weird to me also, but perhaps is it just when the prop is `Component`? 

There is a convention I'm familiar with where components are capitalised to indicate that e.g. `SomeComponent.tsx`

Reply:
Why pass `() => ReactNode` instead of `ReactNode`?